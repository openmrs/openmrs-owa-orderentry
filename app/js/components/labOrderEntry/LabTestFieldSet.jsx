import React from 'react';
import PropTypes from 'prop-types';
import { testsData } from './labData';
import '../../../css/grid.scss';

class LabTestFieldSet extends React.Component {
  render() {
    const { selectedTests, selectTest } = this.props;
    const selectedTestIds = selectedTests.map(test => test.id);
    return (
      <fieldset className="fieldset">
        <legend>Tests</legend>
        <div className="flex-row">
          {
            testsData.map((item, index) => (
              <button
                type="button"
                id="category-test-button"
                key={`${item.id}`}
                className={`lab-tests-btn ${(selectedTestIds.includes(item.id)) ? 'active' : ''}`}
                onClick={() => selectTest(item)}
              >
                {item.test}
              </button>
            ))
          }
        </div>
      </fieldset>
    );
  }
}

LabTestFieldSet.propTypes = {
  selectedTests: PropTypes.array.isRequired,
  selectTest: PropTypes.func.isRequired,
};

export default LabTestFieldSet;
