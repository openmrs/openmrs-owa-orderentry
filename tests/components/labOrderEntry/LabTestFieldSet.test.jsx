import React from 'react';
import LabTestFieldSet from '../../../app/js/components/labOrderEntry/LabTestFieldSet';

let props;
let mountedComponent;
props = {
  selectedTests: [
    { id: 1, test: 'Hemoglobin' },
    { id: 2, test: 'Hematocrit' },
    { id: 3, test: 'blood' },
  ],
  selectTest: () => jest.fn()
};

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<LabTestFieldSet { ...props } />);
  }
  return mountedComponent;
};

describe('Component: LabTestFieldSet', () => {
  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should mount initially', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should support click for each button rendered', () => {
    const component = getComponent();
    const testButton = component.find('#category-test-button').at(0); // click the second button
    testButton.simulate('click', {});
    expect(component).toMatchSnapshot();
  });
});
