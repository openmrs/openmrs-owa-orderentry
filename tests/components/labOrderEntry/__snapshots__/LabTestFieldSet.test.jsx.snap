// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LabTestFieldSet should mount initially 1`] = `
<fieldset
  className="fieldset"
>
  <legend>
    Tests
  </legend>
  <div
    className="tests-box"
  >
    <button
      className="lab-tests-btn active"
      id="category-test-button"
      key="asampleduuid1-234"
      onClick={[Function]}
      type="button"
    >
      samplea
    </button>
    <button
      className="lab-tests-btn active"
      id="category-test-button"
      key="defmpleduuid1-566"
      onClick={[Function]}
      type="button"
    >
      sampleb
    </button>
    <button
      className="lab-tests-btn "
      id="category-test-button"
      key="5sampleduuid2-788"
      onClick={[Function]}
      type="button"
    >
      samplec
    </button>
  </div>
</fieldset>
`;

exports[`Component: LabTestFieldSet should support click for each button rendered 1`] = `
<fieldset
  className="fieldset"
>
  <legend>
    Tests
  </legend>
  <div
    className="tests-box"
  >
    <button
      className="lab-tests-btn active"
      id="category-test-button"
      key="asampleduuid1-234"
      onClick={[Function]}
      type="button"
    >
      samplea
    </button>
    <button
      className="lab-tests-btn active"
      id="category-test-button"
      key="defmpleduuid1-566"
      onClick={[Function]}
      type="button"
    >
      sampleb
    </button>
    <button
      className="lab-tests-btn "
      id="category-test-button"
      key="5sampleduuid2-788"
      onClick={[Function]}
      type="button"
    >
      samplec
    </button>
  </div>
</fieldset>
`;
